pkg/3scale/amp/operator/system_options_provider.go

```diff
@@ -378,11 +378,8 @@ func (s *SystemOptionsProvider) setBackendOptions() error {
 		return fmt.Errorf("'%s' field of '%s' secret must have 'scheme://user:password@host/path' format", component.BackendSecretBackendListenerServiceEndpointFieldName, component.BackendSecretBackendListenerSecretName)
 	}
 
-	if urlObj.Path == "" {
-		urlObj.Path = "/internal/"
-	}
-
-	s.options.BackendServiceEndpoint = urlObj.String()
+	urlObj.Path += "/internal/"
+	s.options.CoreInternalAPIEndpoint = urlObj.String()
```

Comment:
Be my guest

With the current implementation in this PR, there are three env vars
* APICAST_BACKEND_ROOT_ENDPOINT -> `https://backend-3scale.eguzki-3scale.apps.example.com`
* `THREESCALE_CORE_INTERNAL_API` -> `http://backend-listener:3000/internal`
* `BACKEND_PUBLIC_URL` -> `https://backend-3scale.eguzki-3scale.apps.example.com`

APICAST_BACKEND_ROOT_ENDPOINT and BACKEND_PUBLIC_URL will always have the same value. For now at least.  Seems duplicated, shall we simplify it?

For me it seems like we need two of them:

* BACKEND_PUBLIC_URL -> https://backend-3scale.eguzki-3scale.apps.example.com
* BACKEND_INTERNAL_URL -> http://backend-listener:3000/internal

Reply:
In system we still need the backend internal URL **without** the `/internal` prefix, more info on that in [this comment](https://github.com/3scale/3scale-operator/pull/821#issuecomment-1542065989).

As for `APICAST_BACKEND_ROOT_ENDPOINT` and `BACKEND_PUBLIC_URL` indeed the values are duplicated, and I also thought about whether we need to merge them into one.

I think there could potentially be a scenario where the customer would want to have different values.

`BACKEND_PUBLIC_URL` would be used by Active Docs (so that the customer can make requests to the Service Management API from the browser)

`APICAST_BACKEND_ROOT_ENDPOINT` is used by APIcast (so that APIcast can make requests to the Service Management API)

So, depending on where APIcast is deployed (if it's outside of OCP), maybe these two values could be different (within private network vs public).

On the other hand, we could indeed make it simple, and just use one value, and if somebody needs to distinguish these two, they can open an RFE.

Reply:
For now, I maintained both `APICAST_BACKEND_ROOT_ENDPOINT` and `BACKEND_PUBLIC_URL`, even though they have the same value.